src/RendezvousTracker.ts

```diff
@@ -0,0 +1,268 @@
+
+            // Check the clint path map for the corrected file name
+            if (this.clientPathsMap[fileNameAsBrs]) {
+                fileName = fileNameAsBrs;
+            } else if (this.clientPathsMap[fileNameAsXml]) {
+                fileName = fileNameAsXml;
+            }
+        }
+
+        if (!this.clientPathsMap[fileName]) {
+            // Add new file to client path map
+            if (fileNameAsBrs || fileNameAsXml) {
+                // File name did not have a valid extension
+                // Check for both the .brs and .xml versions of the file starting with .brs
+                fileNameAsBrs = this.convertDebuggerPathToClient(fileNameAsBrs);
+                if (fileNameAsBrs) {
+                    fileName = fileNameAsBrs;
+                } else {
+                    fileNameAsXml = this.convertDebuggerPathToClient(fileNameAsXml);
+                    if (fileNameAsXml) {
+                        fileName = fileNameAsXml;
+                    }
+                }
+            }
+
+            this.clientPathsMap[fileName] = {
+                clientPath: this.convertDebuggerPathToClient(fileName),
+                clientLines: {
+                    [lineNumber]: this.convertDebuggerLineToClientLine(fileName, parseInt(lineNumber))
+                }
+            };
+        } else if (!this.clientPathsMap[fileName].clientLines[lineNumber]) {
+            // Add new client line to clint path map
+            this.clientPathsMap[fileName].clientLines[lineNumber] = this.convertDebuggerLineToClientLine(fileName, parseInt(lineNumber));
+        }
+
+        return fileName;
+    }
+
+    /**
+     * Helper function to assist in the creation of a RendezvousLineInfo
+     * @param fileName processed file name
+     * @param lineNumber occurrence line number
+     * @param duration how long the rendezvous took to complete, if not supplied it is assumed to be zero
+     */
+    private createLineObject(fileName: string, lineNumber: number, duration?: string): RendezvousLineInfo {
+        return {
+            clientLineNumber: lineNumber,
+            clientPath: this.clientPathsMap[fileName].clientPath,
+            hitCount: 1,
+            totalTime: this.getTime(duration),
+            type: 'lineInfo'
+        };
+    }
+
+    /**
+     * Helper function to convert the duration to a float or return 0.00
+     * @param duration how long the rendezvous took to complete, if not supplied it is assumed to be zero
+     */
+    private getTime(duration?: string): number {
+        return duration ? parseFloat(duration) : 0.000;
+    }
+
+    /**
+     * Replaces the last occurrence of a string in a string
+     * @param base String to search within
+     * @param replace String to replace
+     * @param replacement What to replace the final occurrence with
+     */
+    private replaceLastStringOccurrence(base: string, replace: string, replacement: string): string {
```

Comment:
Here's a node module for this functionality, so try using that isntead: https://www.npmjs.com/package/replace-last

If that doesn't work, then at least move this function into `util.ts`.

Reply:
Figured there was no need for a import if we could simply write the one function our selfs. I can move to utils for sure.

Reply:
I don't mind using additional modules because they reduce the amount of code we need to maintain ourselves. The one I linked to above has zero dependencies.

Reply:
Ah, good point. Ok I will move over to the module.

Reply:
Done